maven-plugin/src/main/java/org/openmrs/maven/plugins/Build.java

```diff
@@ -0,0 +1,171 @@
+package org.openmrs.maven.plugins;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.io.xpp3.MavenXpp3Writer;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.shared.invoker.MavenInvocationException;
+import org.openmrs.maven.plugins.model.Server;
+import org.openmrs.maven.plugins.utility.Project;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+
+/**
+ * @goal build
+ * @requiresProject false
+ */
+public class Build extends AbstractTask {
+
+    /**
+     * @parameter expression="${serverId}"
+     */
+    private String serverId;
+
+    public Build(){}
+
+    public Build(AbstractTask other, String serverId){
+        super(other);
+        this.serverId = serverId;
+    }
+
+    @Override
+    public void executeTask() throws MojoExecutionException, MojoFailureException {
+        Server server = Server.loadServer(serverId);
+
+
+        File tempFolder = createTempReactorProject(server);
+        cleanInstallServerProject(tempFolder.getPath());
+        deleteTempReactorProject(tempFolder);
+        try {
+            deployWatchedModules(server);
+        } catch (MavenInvocationException e) {
+            throw new MojoFailureException("Failed to deploy watched modules", e);
+        }
+
+    }
+
+    /**
+     * Creates temporary Maven reactor project to build dependencies in the correct order
+     *
+     * @param server
+     * @return
+     * @throws MojoFailureException
+     * @throws MojoExecutionException
+     */
+    private File createTempReactorProject(Server server) throws MojoFailureException, MojoExecutionException {
+        File tempFolder = new File(server.getServerDirectory(), "temp-project");
+        if (!tempFolder.exists()) {
+            tempFolder.mkdir();
+        }
+        Model tempModel = createModel();
+        Set<Project> watchedModules = server.getWatchedProjects();
+        for(Project module: watchedModules){
+            Path newLink = Paths.get(new File(tempFolder, module.getArtifactId()).getAbsolutePath());
+            Path existingfile = Paths.get(module.getPath());
+            try {
+                Files.createSymbolicLink(newLink, existingfile);
```

Comment:
It seems createSymbolicLink requires administrator privileges on Windows. We should catch IOException here and simply copy directory if creating symbolic link fails...